wsmaster/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/model/impl/AgentImpl.java

```diff
@@ -44,10 +44,10 @@ public AgentImpl(String id,
         this.name = name;
         this.version = version;
         this.description = description;
-        this.dependencies = dependencies;
-        this.properties = properties;
+        this.dependencies = dependencies != null ? dependencies : new ArrayList<>();
```

Comment:
getters are doing this check so it looks odd to have it at both places
if it's immutable it should be Collection.emptyList()